docker-compose.yml

```diff
@@ -77,46 +83,92 @@ services:
     ports:
       - "8080:8080"
 
+  # C/C++ functions
   cpp:
     image: ${CPP_CLI_IMAGE}
     working_dir: /code/cpp
     stdin_open: true
     tty: true
-    depends_on:
-      - worker
-      - upload
     volumes:
       - ./clients/cpp:/code/cpp
+      - ./dev/faasm-local/:${FAASM_LOCAL_MOUNT}
 
+  # Python functions
   python:
     image: ${PYTHON_CLI_IMAGE}
     working_dir: /code/python
     stdin_open: true
     tty: true
-    depends_on:
-      - worker
-      - upload
     volumes:
       - ./clients/python:/code/python
+      - ./dev/faasm-local/:${FAASM_LOCAL_MOUNT}
+
+  # Faabric dev
+  faabric:
+    image: ${FAABRIC_CLI_IMAGE}
+    working_dir: /code/faabric
+    stdin_open: true
+    tty: true
+    depends_on:
+      - redis-state
+      - redis-queue
+      - minio
+    environment:
+      - LOG_LEVEL=debug
+      - REDIS_STATE_HOST=redis
+      - REDIS_QUEUE_HOST=redis
+    volumes:
+      - ./faabric:/code/faabric
+      - ./dev/faabric/build/:/build/faabric
 
-  # --- Local cluster dev ---
-  # The following containers are only needed for local cluster dev
+  # Faasm dev
   faasm-cli:
     image: ${FAASM_CLI_IMAGE}
     working_dir: /usr/local/code/faasm
     stdin_open: true
     tty: true
+    privileged: true
     depends_on:
-      - worker
+      - redis-state
+      - redis-queue
+      - minio
     environment:
       - UPLOAD_HOST=${UPLOAD_HOST:-upload}
       - INVOKE_HOST=${INVOKE_HOST:-nginx}
-      - LOG_LEVEL=info
+      - LOG_LEVEL=debug
       - REDIS_QUEUE_HOST=redis-queue
       - REDIS_STATE_HOST=redis-state
       - LD_LIBRARY_PATH=/build/faasm/third-party/lib:/usr/local/lib
     volumes:
       - ./:/usr/local/code/faasm/
-      - ./dev/faasm/build/:/build/faasm/
-      - ./dev/faasm-local:/usr/local/faasm
+      - ./dev/faasm/build/:${FAASM_BUILD_MOUNT}
+      - ./dev/faasm-local/:${FAASM_LOCAL_MOUNT}
 
+  # Distributed tests server
+  dist-test-server:
+    image: ${FAASM_CLI_IMAGE}
+    working_dir: /build/faasm
+    depends_on:
+      - redis-queue
+      - redis-state
+      - minio
+    environment:
+      - LOG_LEVEL=debug
```

Comment:
Although this is run in the CI build, so should be `info`, the dist tests are still relatively unstable, and having `debug` has proven very useful (plus there aren't very many tests actually running).